File: LeetCode/C#/268. Missing Number.cs
public class Solution {
    public int MissingNumber(int[] nums) {
        if(nums.Length == 1)
        {
            if(nums[0] == 0)
            {
                return 1;
            }
            else
            {
                return 0;
            }
        }
        
        var sortList = new SortedList<int, int>();
        
        foreach(var num in nums)
        {
            sortList.Add(num, 1);
        }
        
        var missingNum = nums.Length;
        var targetKey = 0;
        
        foreach(var entry in sortList)
        {
            if(entry.Key == targetKey)
            {
                targetKey++;
            }
            else
            {
                missingNum = targetKey;
                break;
            }
        }
        
        return missingNum;
    }
}

File: LeetCode/Java/268. Missing Number.java
class Solution {
    public int missingNumber(int[] nums) {
        if(nums.length == 1) return 1 - nums[0];

        int xor1 = 0;
        int xor2 = 0;

        for (int i = 0; i < nums.length; i++) {
            xor1 = xor1 ^ (i + 1);
            xor2 = xor2 ^ nums[i];
        }

        return xor1 ^ xor2;
    }
}